authz.go

```diff
@@ -131,8 +152,13 @@ func (a *authorizer) authorize(ctx context.Context, sr authz.SubjectAccessReview
 	}
 	internal := "internal setup error."
 	var via string
+	var client *client
 	for _, check := range checks {
-		client, err := a.client(ctx)
+		if a.AthenzX509 != nil {
```

Comment:
AthenzX509 is a func type, what are we checking for nil here?

Reply:
If a struct property `AthenzX509 ` is set here as a `func` then it should use x509 mode, else it would fallback on token mode.

Reply:
makes sense, thanks